src/ray/core_worker/transport/direct_actor_task_submitter.cc

```diff
@@ -110,19 +110,21 @@ Status CoreWorkerDirectActorTaskSubmitter::SubmitTask(TaskSpecification task_spe
     // Do not hold the lock while calling into task_finisher_.
     task_finisher_.MarkTaskCanceled(task_id);
     rpc::ErrorType error_type;
-    std::unique_ptr<rpc::RayErrorInfo> error_info = nullptr;
+    rpc::RayErrorInfo error_info;
     {
       absl::MutexLock lock(&mu_);
-      auto queue = client_queues_.find(task_spec.ActorId());
-      auto &death_cause = queue->second.death_cause;
-      error_type = GenErrorTypeFromDeathCause(death_cause.get());
-      error_info = GetErrorInfoFromActorDeathCause(death_cause.get());
+      const auto queue = client_queues_.find(task_spec.ActorId());
+      const std::unique_ptr<rpc::ActorDeathCause> &death_cause =
+          queue->second.death_cause;
+      RAY_CHECK(death_cause != nullptr);
```

Comment:
hmm in general we shouldn't crash if the death_cause is unknown?

Reply:
I think it would be better to just make sure it's non-nullable (e.g., pass by ref). The call site shouldn't accept null causes.

Reply:
This requires some refactoring. Right now, this death_cause is for some reasons within the struct, and this has invariant, "if actor is dead this value must be set". What about doing refactoring in a follow up PR?